### PR TITLE
Adding spack recipe for hipsparse

### PR DESCRIPTION
**Pull request by arjun-raj-kuppala**

### CODE REVIEW

File: packages/hipsparse/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Hipsparse(CMakePackage):
+    """hipSPARSE is a SPARSE marshalling library, with multiple supported backends. It sits between the application and a
```
Comment (srekolam) [inline packages/hipsparse/package.py]: the line colums allowed is 79..any thing more i see flake8 errors.Also ensure no trailing whitespaces..

File: packages/hipsparse/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    depends_on('hsakmt-roct@3.5.0', type='build')
+    depends_on('hsa-rocr-dev@3.5.0', type='build')
+    depends_on('hip@3.5.0:', type='build', when='@3.5.0:')
+    depends_on('rocm-device-libs@3.5.0:', type='build', when='@3.5.0')
+    depends_on('comgr@3.5.0:', type='build', when='@3.5.0')
+    depends_on('rocsparse@3.5.0:', type='build', when='@3.5.0')
+
+    def cmake_args(self):
+        args = [
+               '-DCMAKE_CXX_FLAGS=-I {0}/include'.format(self.spec['hsa-rocr-dev'].prefix),
```
Comment (srekolam) [inline packages/hipsparse/package.py]: i think the allowed width is 79. so, split the line. also.. With the args [ ] , i see some issues with indentation which is leading to flake8 errors after we push to spack

**Review by srekolam — CHANGES_REQUESTED** [review]
please address the cosmetic errors otherwise will give flake8 issues later on.. i have still bunch of them open

**Review by arjun-raj-kuppala — COMMENTED** [review]
@srekolam All the flake8 issues are addressed, please review and merge this

**Review by srekolam — APPROVED** [review]

**Pull request merged**